src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -692,25 +692,29 @@ private BoundExpression BindDeclarationVariables(TypeSymbol declType, VariableDe
                         var subExpressions = builder.ToImmutableAndFree();
 
                         var uniqueFieldNames = PooledHashSet<string>.GetInstance();
-                        RemoveDuplicateInferredTupleNames(namesBuilder, uniqueFieldNames);
+                        if (RemoveDuplicateInferredTupleNamesAndReportEmptied(namesBuilder, uniqueFieldNames))
+                        {
+                            namesBuilder.Free();
+                            namesBuilder = null;
+                        }
                         uniqueFieldNames.Free();
 
-                        var tupleNames = namesBuilder.ToImmutableAndFree();
-                        var inferredPositions = tupleNames.SelectAsArray(n => n != null);
+                        var tupleNames = namesBuilder is null ? default : namesBuilder.ToImmutableAndFree();
+                        var inferredPositions = tupleNames.IsDefault ? default : tupleNames.SelectAsArray(n => n != null);
```

Comment:
Is it worth extracting a helper method for this block that returns `tupleNames` and `inferredPositions`?

Reply:
For just those two lines, I don't think it's worth it.
There's actually a block above and below that looks very similar between both methods, but enough differences that the variables entering the region are too many (imo).